### PR TITLE
modified apis to support prids and batch

### LINKED ISSUE #23: [graphSDK] Standardize API to support PRI
# Problem
Standardize and modify APIs to support PRI calculation workflows

## Acceptence Criterea
- APIs should be able to allow PRI **verification** and **generation** for each page in Private Friendship graph
- Expose which connections require public keys

### PR DESCRIPTION
**Pull request by aramikm**
# Goal
The goal of this PR is modify apis to support PRID usecase and allow batch processing

Closes #23 

# Discussion
- Tests are not fixed yet
- Batch apis would need transactional support

# Checklist
- [x] Tests added

### CODE REVIEW

File: core/src/api/api.rs [hunk 1]
```diff
@@ -135,107 +134,111 @@ impl GraphAPI for GraphState {
...
+				false => vec![],
+			};
 
-		match (connection_type.privacy_type(), include_secret_keys) {
-			(PrivacyType::Public, _) => graph.import_public(connection_type, pages),
-			(PrivacyType::Private, true) => {
-				graph.import_private(&dsnp_config, connection_type, &pages, resolved_keys)?;
-				self.pri_manager
+			let graph = user_graph.graph_mut(&schema_id);
+			graph.clear();
```
Comment (JoeCap08055) [inline core/src/api/api.rs:170]: Regarding the TODO above to make this transactional, I suppose one step toward that would be to create a new graph and only replace the existing one when the import completes successfully.

File: core/src/graph/key_manager.rs [hunk 2]
```diff
@@ -176,12 +179,27 @@ impl PublicKeyProvider for PublicKeyManager {
 
 		Ok(())
 	}
+
+	fn filter_users_without_keys(&self, dsnp_user_ids: Vec<DsnpUserId>) -> Vec<DsnpUserId> {
```
Comment (JoeCap08055) [inline core/src/graph/key_manager.rs:183]: nit: maybe 'get' or 'find' would be more appropriate naming than 'filter'

File: core/src/api/api.rs (hunk 1 above)
Comment (aramikm) [reply core/src/api/api.rs:170]: We can do that or I was thinking about having a dryrun flag described in https://github.com/LibertyDSNP/graph-sdk/issues/40. Basically we run all the input using **dryrun = true** and if everything looked good we would run it again.

File: core/src/graph/key_manager.rs (hunk 2 above)
Comment (aramikm) [reply core/src/graph/key_manager.rs:183]: will change

File: core/src/api/api.rs (hunk 1 above)
Comment (aramikm) [reply core/src/api/api.rs:170]: Open to both ideas. We can do a quick evaluation of how much is the effort with each one and progress from there. I'll update that ticket to have your approach as an alternative to explore.

**Review by JoeCap08055 — APPROVED** [review]

**Pull request merged**